Adds frontend information in contribution guide (#4877)
rotki/rotki: docs/contribute.rst

The unit tests that are testing functions and components are using ``vitest`` and ``vue-test-utils`` and you can run
them by:

::

    npm run test:unit -w app

These are supposed to be small tests ensuring that parts of the code work good in isolation.

The second type of tests is an `e2e` test suite using ``cypress``. The e2e tests require the python virtual environment
because they depend on the actual python backend. These tests ensure proper e2e functionality and application integration
and try to replicate scenarios of real user interaction through the application.

To run the e2e tests you need to run the following command inside the frontend directory:

::

    npm run test:integration-ci -w app

The above command will run the e2e tests in headless mode. If you want to debug specific tests you can also run:

::

    npm run test:integration -w

This command will open the Cypress Test Runner window where you can select specific suites to execute.

Linting
--------

Before committing and pushing your commits ensure that you fix any lint issues. You can do this by running:

::

    npm run lint:fix

.. note::

>     While lint warning are not fatal and will not fail the CI pipeline it would be better a PR
>     reduces the number of warnings and doesn't introduces new ones. Warnings are things that

Reviewer: ```suggestion
    While lint warning are not fatal and will not fail the CI pipeline it would be better if a PR
    reduces the number of warnings and doesn't introduce new ones. Warnings are things that
```

>     need to be fixed and the will be converted to errors in the future.

Reviewer 2: ```suggestion
    need to be fixed and they will be converted to errors in the future.
```

> 

Reviewer: Too many spaces here?

> 
> 
> 

Reviewer 2: ```suggestion
```
